tpl/collections/collections.go

```diff
@@ -561,21 +568,95 @@ func (ns *Namespace) Union(l1, l2 interface{}) (interface{}, error) {
 		case reflect.Array, reflect.Slice:
 			r := reflect.MakeSlice(l1v.Type(), 0, 0)
 
-			if l1v.Type() != l2v.Type() {
+			if l1v.Type() != l2v.Type() &&
+				l1v.Type().Elem().Kind() != reflect.Interface &&
+				l2v.Type().Elem().Kind() != reflect.Interface {
 				return r.Interface(), nil
 			}
 
+			var l1vv reflect.Value
 			for i := 0; i < l1v.Len(); i++ {
-				elem := l1v.Index(i)
-				if !ns.In(r.Interface(), elem.Interface()) {
-					r = reflect.Append(r, elem)
+				l1vv = l1v.Index(i)
+				if !ns.In(r.Interface(), l1vv.Interface()) {
+					r = reflect.Append(r, l1vv)
 				}
 			}
 
 			for j := 0; j < l2v.Len(); j++ {
-				elem := l2v.Index(j)
-				if !ns.In(r.Interface(), elem.Interface()) {
-					r = reflect.Append(r, elem)
+				l2vv := l2v.Index(j)
+
+				switch l1vv.Kind() {
+				case reflect.String:
+					l2t, err := toString(l2vv)
+					if err == nil && !ns.In(r.Interface(), l2t) {
+						r = reflect.Append(r, reflect.ValueOf(l2t))
+					}
+				case reflect.Int:
+					l2t, err := toInt(l2vv)
+					if err == nil && !ns.In(r.Interface(), l2t) {
+						r = reflect.Append(r, reflect.ValueOf(int(l2t)))
+					}
+				case reflect.Int8:
+					l2t, err := toInt(l2vv)
+					if err == nil && !ns.In(r.Interface(), l2t) {
+						r = reflect.Append(r, reflect.ValueOf(int8(l2t)))
+					}
+				case reflect.Int16:
+					l2t, err := toInt(l2vv)
+					if err == nil && !ns.In(r.Interface(), l2t) {
+						r = reflect.Append(r, reflect.ValueOf(int16(l2t)))
+					}
+				case reflect.Int32:
+					l2t, err := toInt(l2vv)
+					if err == nil && !ns.In(r.Interface(), l2t) {
+						r = reflect.Append(r, reflect.ValueOf(int32(l2t)))
+					}
+				case reflect.Int64:
+					l2t, err := toInt(l2vv)
+					if err == nil && !ns.In(r.Interface(), l2t) {
+						r = reflect.Append(r, reflect.ValueOf(l2t))
+					}
+				case reflect.Float32:
+					l2t, err := toFloat(l2vv)
+					if err == nil && !ns.In(r.Interface(), float32(l2t)) {
+						r = reflect.Append(r, reflect.ValueOf(float32(l2t)))
+					}
+				case reflect.Float64:
+					l2t, err := toFloat(l2vv)
+					if err == nil && !ns.In(r.Interface(), l2t) {
+						r = reflect.Append(r, reflect.ValueOf(l2t))
+					}
+				case reflect.Interface:
```

Comment:
When I read "support interfaces" the last time, I expected a more complete support set than this; i.e. I wanted `*Page`. You may argue that is outside of scope, or the next step or whatever, but to me that is a sign that we have to look at the problem again and see if it can be made simpler/more generic, which would cover more bases.